src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -94,6 +94,59 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         /// Must be a non-null interface property, method, or event.
         /// </param>
         ISymbol FindImplementationForInterfaceMember(ISymbol interfaceMember);
+
+        /// <summary>
+        /// Converts the symbol to a string representation with given nullability.
+        /// </summary>
+        /// <param name="nullability">The nullability to use for the symbol.</param>
+        /// <param name="format">Format or null for the default.</param>
+        /// <returns>A formatted string representation of the symbol.</returns>
+        string ToDisplayString(Nullability nullability, SymbolDisplayFormat format = null);
+
+        /// <summary>
+        /// Convert a symbol to an array of string parts, each of which has a kind, with a given
+        /// nullability. Useful for colorizing the display string.
+        /// </summary>
+        /// <param name="nullability">The nullability to use for formatting.</param>
+        /// <param name="format">Formatting rules - null implies
+        /// SymbolDisplayFormat.ErrorMessageFormat.</param>
+        /// <returns>A read-only array of string parts.</returns>
+        ImmutableArray<SymbolDisplayPart> ToDisplayParts(Nullability nullability, SymbolDisplayFormat format = null);
+
+        /// <summary>
+        /// Convert a symbol to a string that can be displayed to the user. May be tailored to a
+        /// specific location in the source code.
+        /// </summary>
+        /// <param name="semanticModel">Binding information (for determining names appropriate to
+        /// the context).</param>
+        /// <param name="position">A position in the source code (context).</param>
+        /// <param name="nullability">The nullability to use in formatting.</param>
+        /// <param name="format">Formatting rules - null implies
+        /// <returns>A formatted string that can be displayed to the user.</returns>
+        string ToMinimalDisplayString(
+            SemanticModel semanticModel,
+            int position,
+            Nullability nullability,
+            SymbolDisplayFormat format = null);
```

Comment:
this really feels like we're going to be virally spreading this outward.  consdier anything in the IDE that ends up calling helpers like this.  this info needs to pass outwards to any and all callpaths.